packages/editor/src/components/editor-help/test/index.native.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { fireEvent, render, waitForElementToBeRemoved } from 'test/helpers';
+
+/**
+ * Internal dependencies
+ */
+import EditorHelp from '../index';
+
+it( 'lists help topics', async () => {
+	const screen = render( <EditorHelp isVisible /> );
+	const helpTopic = await screen.findByText( 'Customize blocks' );
+
+	expect( helpTopic ).toBeTruthy();
+} );
+
+it( 'navigates to help topic detail screen', async () => {
+	const screen = render( <EditorHelp isVisible /> );
+	const helpTopic = await screen.findByText( 'Customize blocks' );
+	fireEvent.press( helpTopic );
+
+	expect(
+		screen.getByText(
+			'Each block has its own settings. To find them, tap on a block. Its settings will appear on the toolbar at the bottom of the screen.'
+		)
+	).toBeTruthy();
+} );
+
+it( 'navigates back from help topic detail screen', async () => {
+	const screen = render( <EditorHelp isVisible /> );
+	const helpTopic = await screen.findByText( 'Customize blocks' );
+	fireEvent.press( helpTopic );
+
+	const detailScreenTitle1 = screen.getByText(
+		'Each block has its own settings. To find them, tap on a block. Its settings will appear on the toolbar at the bottom of the screen.'
+	);
+	expect( detailScreenTitle1 ).toBeTruthy();
+
+	const backButton = screen.getAllByA11yLabel( 'Go back' );
+	fireEvent.press( backButton[ backButton.length - 1 ] );
+
+	// Currently logs `act` warning due to https://git.io/JYYGE
+	await waitForElementToBeRemoved( () =>
+		screen.getByText(
+			'Each block has its own settings. To find them, tap on a block. Its settings will appear on the toolbar at the bottom of the screen.'
+		)
+	);
+} );
+
+it( 'dismisses when close button is pressed', async () => {
+	const closeMock = jest.fn();
+	const screen = render( <EditorHelp isVisible close={ closeMock } /> );
+	const closeButton = await screen.findByA11yLabel( 'Go back' );
+	fireEvent.press( closeButton );
```

Comment:
> While researching to understand the code here, I came across [this article](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library)

This is a fantastic article! I had not read this. Thank you for sharing. 

> I see `userEvent` being used on the web-side of Gutenberg, but not for mobile and I'm not sure if there's a specific reason for that. I wasn't able to find any resources around its use in React Native specifically, but seems like it may be worthwhile adopting?

I had not seen `@testing-library/user-event` before. Its mission does appear to be very useful and relevant for writing better tests. That said, I do not believe we can leverage it today with our native tests, it does not appear there is support for—or event plans to support—React Native. If that support is introduced at some point, I think we should explore/consider leveraging the library.

Reply:
I did find https://github.com/callstack/react-native-testing-library/pull/674, which attempts to make RNTL's `fireEvent.press` more aligned with the approach taken by `@testing-library/user-event`.

Reply:
Ah, I see! I wasn't fully grokking the separation between React Testing Library and Native Testing Library. Thank you for sharing, I'll follow along with that GitHub issue and will bookmark Callstack's repository for future reference. :)